bot/utils/dma_api.py

```diff
@@ -47,31 +47,54 @@ async def wrapped(search):
     return wrapper
 
 
-@get_api()
-async def get_article(number, articles):
-    return [
-        item
-        for item in articles
-        if item["series"] == str(number)[0]
-        and item["category"] == str(number)[1]
-        and item["digit"] == str(number)[2]
-    ]
-
-
-@get_api()
-async def get_series(series, articles):
-    return [item for item in articles if item["series"] == str(series)]
+async def get_search_api(query_string: str = None, series_number: int = None):
+    query = "?query={" + re.sub(DMA_QUERY_REGEX, "", query_string) + "}" if query_string is not None else ""
+    series = "?section=" + DMA_SERIES_IDS[series_number] if series_number is not None else ""
+    request_url = f'{DMA_API_SEARCH_URL}{query}{series}'
+    try:
+        async with aiohttp.ClientSession() as sess:
+            response = await sess.get(request_url)
+            response.raise_for_status()
+            data = await response.json()
+
+            results = data["results"]
+            results_parse = []
+
+            for item in results:
+                if re.search(DMA_TITLE_REGEX, item["name"]):
+                    article = {}
+                    # Article
+                    number, name = item["name"].split(": ")
+                    article["series"] = number[0]
+                    article["category"] = number[1]
+                    article["digit"] = number[2]
+                    article["name"] = name
+                    soup = BeautifulSoup(item["body"], "html.parser")
+
+                    placeholder = f"... [continue reading](https://dis.gd/dma{number})"
+                    article["text"] = textwrap.shorten(
+                        soup.p.text, 800, placeholder=placeholder
+                    )
+                    results_parse.append(article)
+
+            results_parse = sorted(results_parse, key=lambda x: (x["series"], x["category"], x["digit"]))
+    except Exception:
+        logger.exception("Error fetching information from DMA")
+    return results_parse
```

Comment:
We should probably still return something here for the user.

Right now this will get a bit weird because an error log will be logged and then it will generate a fatal error because results_parse will not be defined (assuming a request failure).

You could use try/except/else here:
```py
try:
   # something
except Exception as e:
   # handle error
else:
   # run this *only* if there were no errors
```

Reply:
From testing the /search endpoint, seems like the only error thrown is when there are no headers. Unless someone is client modding to send a value outside of the given range of 1-5 for the series, this should all be fine after moving the definition of results_parse to outside the try/except.